# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.h
# Commit: 66c3420f8f7f
# Full Hash: 66c3420f8f7fc12124e9fd1825d1d1fd8776b022
# Author: stransky <stransky@redhat.com>
# Date: 2022-07-26 21:44:38
# Regressor Bug: 1777664
# File Overlap Count: 1
# Description:
#   Bug 1777664 [Wayland] Remove EnableRenderingToWindow()/DisableRenderingToWindow/()/ResumeCompositorHiddenWindow() r=emilio
#   
#   Enable composition only once when nsWindow is created and don't stop compositon until a window is destroyed.
#   Remove COMPOSITOR_PAUSED_MISSING_WINDOW as we don't manage compositing for hidden windows any more.
#   
# ==============================================================================

diff -r 0531e0fd8b47 -r 66c3420f8f7f widget/gtk/nsWindow.h
--- a/widget/gtk/nsWindow.h	Tue Jul 26 14:56:47 2022 +0000
+++ b/widget/gtk/nsWindow.h	Tue Jul 26 14:56:48 2022 +0000
@@ -434,9 +434,6 @@
     COMPOSITOR_ENABLED,
     // WebRender compositor is paused after window creation.
     COMPOSITOR_PAUSED_INITIALLY,
-    // WebRender compositor is paused because GtkWindow is hidden,
-    // we can't draw into GL context.
-    COMPOSITOR_PAUSED_MISSING_WINDOW,
     // WebRender compositor is paused as we're repainting whole window and
     // we're waiting for content process to update page content.
     COMPOSITOR_PAUSED_FLICKERING
@@ -483,9 +480,6 @@
 
   void TryToShowNativeWindowMenu(GdkEventButton* aEvent);
 
-  void EnableRenderingToWindow();
-  void DisableRenderingToWindow();
-  void ResumeCompositorHiddenWindow();
   void WaylandStartVsync();
   void WaylandStopVsync();
   void DestroyChildWindows();
